staging/src/k8s.io/controller-manager/pkg/features/featureconst/kube_features.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package featureconst
```

Comment:
I'm not sure that a new package `featureconst` is needed to store the feature constants, just `k8s.io/controller-manager/pkg/features/kube_features` would be enough I think.

Reply:
My understanding is that the init func will be called whenever people import `k8s.io/controller-manager/pkg/feature` and we would like to separate it for the use case like `allow people to depend on teh consts without gettng auto-registration`(suggested by @deads2k ).

Reply:
I believe Davids idea was to allow people to pick up the constants and be able to choose whether to pick up the auto-registration. The idea is to move away from the auto-registration magic. I wonder if it would make more sense to split out the init (auto-registration) into an autoregister package, so that the behavior there is a bit more explicit?

Reply:
Could we do this by having a single package `k8s.io/controller-manager/pkg/feature` and remove `init()`? That way you can consume constants, but registering gates would requier an explicit call to `SetupCurrentKubernetesSpecificFeatureGates`. Or was the separate package added because we want to keep `init()` for registration?

Reply:
If we go the auto-registration route, maybe `k8s.io/controller-manager/pkg/features/register` might be cleaner?

Reply:
That sounds good. Updated to use `k8s.io/controller-manager/pkg/features/register`. Thank you 😊